zerver/tests/test_retention.py

```diff
@@ -895,7 +895,7 @@ def test_do_delete_messages_multiple(self) -> None:
         with queries_captured() as queries:
             do_delete_messages(realm, messages)
         self.assertFalse(Message.objects.filter(id__in=message_ids).exists())
-        self.assert_length(queries, 37)
+        self.assert_length(queries, 18)
```

Comment:
This should be in an earlier commit; I'll fix as I'm merging the first couple commits.